fix typos error in handlers_test.go file

shikanon opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
fix typos error to make it better: 
-  to change variable `fakeHttp` to `fakeHTTPGetter` make it conforms to golint specification.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
xref  #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs
NONE
```

shikanon committed:
fix typos error in pkg/kubelet/lifecycle

shikanon commented:
/assign @vishh

mattjmcnaughton reviewed (changes requested):
Thanks for your pr and welcome to k8s :)

In Golang, changing the capitalization changes the visibility of the `type`. In other words, changing `fakeHTTP` to `FakeHTTP` makes the struct public. I don't think we want `FakeHTTP` to be public here, as its only used in this testing context.

Instead, I wonder if we could leave the struct as `fakeHTTP` and change the variable name to something other than `fakeHTTP`? I understand we don't want to use `fakeHttp` because the `TTP` should be capitalized. But maybe we can add extra descriptiveness to the variable name?

Also, are these the only Golint errors for the package? If yes, we can remove the package from `hack/.golint_failures`.

mattjmcnaughton commented:
/ok-to-test

Also, can you please update your pull request to say that you `xref # ISSUE` instead of `Fix # ISSUE`. Using `Fix` will close the issue, which we do not want as there are still golint failures we need to address.

shikanon commented:
> Thanks for your pr and welcome to k8s :)
> 
> In Golang, changing the capitalization changes the visibility of the `type`. In other words, changing `fakeHTTP` to `FakeHTTP` makes the struct public. I don't think we want `FakeHTTP` to be public here, as its only used in this testing context.
> 
> Instead, I wonder if we could leave the struct as `fakeHTTP` and change the variable name to something other than `fakeHTTP`? I understand we don't want to use `fakeHttp` because the `TTP` should be capitalized. But maybe we can add extra descriptiveness to the variable name?
> 
> Also, are these the only Golint errors for the package? If yes, we can remove the package from `hack/.golint_failures`.

hi, @mattjmcnaughton 

Thanks for your advice and I'm very happy to join this :)

Your suggestion is nice, to use other vocabulary instead of `fakeHttp` is better, like `fakeHTTPGetter` or `fakeHTTPTester`. I fix it.

The package in `hack/.golint_failures` can not remove, beause I only fix `handlers_test.go` and many golint error in this package. 

This is my first commit for this project. Thank you for your help.

shikanon commented:
hi @mattjmcnaughton, I find the k8s-ci-robot returns errors of  all failed tests [pull-kubernetes-e2e-gce](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88450/pull-kubernetes-e2e-gce/1231993056341790724/) and [pull-kubernetes-e2e-gce-100-performance](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88450/pull-kubernetes-e2e-gce-100-performance/1231993056396316672/) have nothing to do with my changed codes.  And what can I do next?

shikanon commented:
/retest

mattjmcnaughton reviewed (approved):
/lgtm

Perfect, thank you for your changes :)

One final request - can you update the PR description so that its accurate given the most recent round of changes? Currently it still suggests that we updated the struct name.

mattjmcnaughton commented:
/assign @tallclair

shikanon commented:
> /lgtm
> 
> Perfect, thank you for your changes :)
> 
> One final request - can you update the PR description so that its accurate given the most recent round of changes? Currently it still suggests that we updated the struct name.

Thank you for your reminder. I fixed it :)

shikanon commented:
hi, @ahg-g @draveness @dims 
It is simple typo, PTAL :)

draveness commented on pkg/kubelet/lifecycle/handlers_test.go:
```diff
@@ -133,8 +133,8 @@ func (f *fakeHTTP) Get(url string) (*http.Response, error) {
 func TestRunHandlerHttp(t *testing.T) {
-	fakeHttp := fakeHTTP{}
-	handlerRunner := NewHandlerRunner(&fakeHttp, &fakeContainerCommandRunner{}, nil)
+	fakeHTTPGetter := fakeHTTP{}
```
Thanks for the catch but could we remove this package from the go lint files?

draveness replied:
OK, I see the comments above

shikanon replied:
I only fix one file and there are many golint errors in this package.

shikanon commented:
ping @dchen1107 @derekwaynecarr @yujuhong @tallclair  for  approved

shikanon commented:
/assign @dchen1107 @derekwaynecarr @yujuhong @tallclair

yujuhong commented:
/approve
